src/lib/Component/MenuComponent.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\TwigComponents\Component;
+
+use Ibexa\Contracts\TwigComponents\ComponentInterface;
+use Twig\Environment;
+
+/**
+ * This Twig Component allows to render menu.
+ */
+class MenuComponent implements ComponentInterface
```

Comment:
Use
```yaml
lazy: Ibexa\Contracts\TwigComponents\ComponentInterface
```
if needed.

```suggestion
final class MenuComponent implements ComponentInterface
```

Reply:
`Ibexa\Bundle\TwigComponents\DependencyInjection\IbexaTwigComponentsExtension::registerConfiguredComponents` sets services created from configuration as lazy by default.

Reply:
You can add
```php
$definition->addTag('proxy', ['interface' => ComponentInterface::class]);
```
to allow those services to be `final` and lazy at the same time.
It will also cause proxies to be generated against the interface only.

See https://symfony.com/doc/5.x/service_container/lazy_services.html#interface-proxifying.

Reply:
Nice one, but please let me introduce this change in a separate PR.